src/controllers.rs

```diff
@@ -31,7 +32,7 @@ mod prelude {
             crate::util::json_response(t)
         }
 
-        fn query(&self) -> HashMap<String, String> {
+        fn query(&self) -> IndexMap<String, String> {
```

Comment:
`IndexMap` defaults to the `RandomState` hasher from `std`, so this looks fine from a DOS perspective.